Sources/LiveKit/Core/Room+EngineDelegate.swift

```diff
@@ -207,13 +207,26 @@ extension Room {
             return
         }
 
-        let segments = packet.segments.map { $0.toLKType() }
+        let segments = packet.segments.map { segment in
+            TranscriptionSegment(id: segment.id,
+                                 text: segment.text,
+                                 language: segment.language,
+                                 firstReceivedTime: _state.transcriptionReceivedTimes[segment.id] ?? Date(),
+                                 lastReceivedTime: Date(),
+                                 isFinal: segment.final)
+        }
 
         guard !segments.isEmpty else {
             log("[Transcription] Received segments are empty", .warning)
             return
         }
 
+        _state.mutate { state in
+            for segment in segments {
+                state.transcriptionReceivedTimes[segment.id] = segment.firstReceivedTime
```

Comment:
are we sure we want to keep all received transcription segements in memory indefinitely? 
For the components implementation I added an option to configure how many segments should be kept at a maximum before dropping the oldest ones in favor of newly arriving ones.

Reply:
Good point - we can probably just drop them as soon as `final` comes over and then skip the configurable limits. I'll update it. (FWIW - this implementation is pretty lean as-is because it keeps only the timestamps. storing the segments themselves is left to the client or components framework)

Reply:
i changed it so they're wiped when final